### PR TITLE
Improve rosetta Dockerfile and workflow

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

- Change rosetta all-in-one Dockerfile to build from local source
- Change rosetta validation job to build docker image with project root as context

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Rosetta validation keeps failing for #1944 but it runs fine locally. The failure is caused by a bug introduced in #1944 and the rosetta process keeps crashing. When I ran it locally, I didn't specify the feature branch so the image is built from master which has no issue.

The PR changes the Dockerfile to always build from the local source tree to prevent such inconsistency.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-rosetta/build/Dockerfile [hunk 1]
```diff
@@ -54,36 +46,32 @@ RUN ln -s /data/db/main /var/lib/postgresql/9.6/main
...
-COPY --from=rosetta-builder /tmp/hedera-mirror-node/hedera-mirror-rosetta/config/application.yml ./config/application.yml
+COPY --from=rosetta-builder /hedera-mirror-rosetta/rosetta-executable .
+COPY --from=rosetta-builder /hedera-mirror-rosetta/config/application.yml ./config/application.yml
 
 # Copy the Importer Jar and Config from the Java-Builder stage
 WORKDIR /var/importer
 COPY --from=java-builder /hedera-mirror-node/hedera-mirror-importer/target/hedera-mirror-importer-*exec.jar ./hedera-mirror-importer.jar
 
-WORKDIR /hedera-mirror-node/hedera-mirror-rosetta/build
+COPY hedera-mirror-rosetta/build /build
```
Comment (steven-sheehy) [inline hedera-mirror-rosetta/build/Dockerfile:73]: This should have a `--from=rosetta-builder ` right? I don't see anywhere else the rosetta build is copied.

Comment (xin-hedera) [reply hedera-mirror-rosetta/build/Dockerfile:73]: the build directory has the entrypoint script and the supervisord config files, they are static so can just copy from local.

The rosetta binary is copied from `rosetta-builder` in a different step

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**